File: modelo/examen.cs
using System;
using System.Data;
public class examen : oObjeto
{

    public DateTime Fecha { set; get; }
    public string Observaciones { set; get; }
    public int Idmateria { set; get; }
    public string Tipo { set; get; }
    public int Idciclolectivo { set; get; }

    public examen() { }

    public examen(DataRow dr)
    {
        this.Id = Convert.ToInt32(dr["id"]);
        this.Fecha = Convert.ToDateTime(dr["fecha"]);
        this.Observaciones = dr["observaciones"].ToString();
        this.Idmateria = Convert.ToInt32(dr["idmateria"]);
        this.Idciclolectivo = Convert.ToInt32(dr["idciclolectivo"]);
        this.Tipo = dr["tipo"].ToString();
    }

}

File: ClientApp/src/app/clases/examen.ts

export class examen
{
    id: number;
    fecha: Date; 
    idmateria: number;
    idciclolectivo: number;
    tipo: string;
    observaciones: string;

    constructor(obj:any)
    {
        this.id = +obj.id;
        this.fecha = new Date(obj.fecha);
        this.idmateria =+ obj.idmateria;
        this.idciclolectivo = +obj.idciclolectivo;
        this.tipo = obj.tipo;
        this.observaciones = obj.observaciones;
    }
}